File: src/cpp_launch_example/src/twist_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "geometry_msgs/msg/twist.hpp"

class TwistPublisher : public rclcpp::Node
{
public:
    TwistPublisher()
    : Node("twist_publisher"), linear_velocity_(0.0), angular_velocity_(0.5)
    {
        publisher_ = this->create_publisher<geometry_msgs::msg::Twist>("/turtlesim1/turtle1/cmd_vel", 10);
        timer_ = this->create_wall_timer(
            std::chrono::milliseconds(100), std::bind(&TwistPublisher::timer_callback, this));
        RCLCPP_INFO(this->get_logger(), "Publishing Twist messages");
    }

private:
    void timer_callback()
    {
        auto msg = geometry_msgs::msg::Twist();
        linear_velocity_ += 0.005;  // Incremento lineal por ciclo de temporizador
        msg.linear.x = linear_velocity_;
        msg.angular.z = angular_velocity_;
        publisher_->publish(msg);
        RCLCPP_INFO(this->get_logger(), "Publishing: linear=%.3f, angular=%.3f", linear_velocity_, angular_velocity_);
    }

    rclcpp::Publisher<geometry_msgs::msg::Twist>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
    double linear_velocity_;
    const double angular_velocity_;
};

int main(int argc, char * argv[])
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<TwistPublisher>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/py_launch_example/py_launch_example/twist_publisher.py
# py_launch_example/scripts/twist_publisher.py
import rclpy
from rclpy.node import Node
from geometry_msgs.msg import Twist

class TwistPublisher(Node):
    def __init__(self):
        super().__init__('twist_publisher')
        self.publisher_ = self.create_publisher(Twist, '/turtlesim1/turtle1/cmd_vel', 10)
        self.timer_period = 0.1  # seconds
        self.linear_increase = 0.005  # Incremento lineal por ciclo de temporizador
        self.linear_velocity = 0.0
        self.angular_velocity = 0.6  # Velocidad angular constante
        self.timer = self.create_timer(self.timer_period, self.timer_callback)
        self.get_logger().info('Publishing Twist messages')

    def timer_callback(self):
        msg = Twist()
        self.linear_velocity += self.linear_increase  # Incrementar la velocidad lineal gradualmente
        msg.linear.x = self.linear_velocity
        msg.angular.z = self.angular_velocity
        self.publisher_.publish(msg)
        self.get_logger().info(f'Publishing: linear={self.linear_velocity}, angular={self.angular_velocity}')

def main(args=None):
    rclpy.init(args=args)
    node = TwistPublisher()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()
